jdbc/jooq/src/main/java/example/springdata/jdbc/jooq/JooqMethods.java

```diff
@@ -0,0 +1,46 @@
+package example.springdata.jdbc.jooq;
+
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static example.springdata.jdbc.basics.simpleentity.domain.tables.Category.CATEGORY;
+
+/**
+ * Implementations for custom repository access using jOOQ.
+ *
+ * @author Florian Lüdiger
+ */
+@Component
+public class JooqMethods {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    private final DSLContext dslContext;
+
+    public JooqMethods(JdbcTemplate jdbcTemplate, DSLContext dslContext) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.dslContext = dslContext;
+    }
+
+    List<Category> getCategoriesWithAgeGroup(AgeGroup ageGroup) {
+        Result<Record> res = this.dslContext.select().from(CATEGORY).where(CATEGORY.AGE_GROUP.equal(ageGroup.name())).fetch();
+        List<Category> categoryList = new ArrayList<Category>();
```

Comment:
Can we use this for the conversion? https://www.jooq.org/javadoc/3.9.x/org/jooq/Result.html#into-java.lang.Class-

Reply:
We can and I think it's a very elegant solution - done